src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -326,14 +326,15 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
     const SchedulingKey &scheduling_key, const rpc::Address *raylet_address) {
   auto &scheduling_key_entry = scheduling_key_entries_[scheduling_key];
 
-  if (scheduling_key_entry.pending_lease_requests.size() ==
-      max_pending_lease_requests_per_scheduling_category_) {
+  uint64_t max_pending_lease_requests =
+      std::max(get_max_pending_lease_requests_per_scheduling_category_(),
```

Comment:
I think we don't need max? (just calling get_max_pending_lease_requests_per_scheduling_category_() looks sufficient). Or alternatively we can apply the minimal lease requests per category here?